Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/PostgresCentralRepoSettings.java

```diff
@@ -377,95 +305,72 @@ public void setPort(int port) throws CentralRepoException {
-            throw new CentralRepoException("Invalid database name. Cannot be empty."); // NON-NLS
-        } else if (!Pattern.matches(DB_NAMES_REGEX, dbName)) {
-            throw new CentralRepoException("Invalid database name. Name must start with a lowercase letter and can only contain lowercase letters, numbers, and '_'."); // NON-NLS
-        }
-
-        this.dbName = dbName.toLowerCase();
+        connSettings.setDbName(dbName);
     }
 
     /**
      * @return the bulkThreshold
      */
     int getBulkThreshold() {
-        return bulkThreshold;
+        return connSettings.getBulkThreshold();
     }
 
     /**
      * @param bulkThreshold the bulkThreshold to set
      */
     public void setBulkThreshold(int bulkThreshold) throws CentralRepoException {
-        if (bulkThreshold > 0) {
-            this.bulkThreshold = bulkThreshold;
-        } else {
-            throw new CentralRepoException("Invalid bulk threshold."); // NON-NLS
-        }
+        connSettings.setBulkThreshold(bulkThreshold);
     }
 
     /**
      * @return the userName
      */
     public String getUserName() {
-        return userName;
+        return connSettings.getUserName();
     }
 
     /**
      * @param userName the userName to set
      */
     public void setUserName(String userName) throws CentralRepoException {
-        if (userName == null || userName.isEmpty()) {
-            throw new CentralRepoException("Invalid user name. Cannot be empty."); // NON-NLS
-        } else if (!Pattern.matches(DB_USER_NAMES_REGEX, userName)) {
-            throw new CentralRepoException("Invalid user name. Name must start with a letter and can only contain letters, numbers, and '_'."); // NON-NLS
-        }
-        this.userName = userName;
+        connSettings.setUserName(userName);
     }
 
     /**
      * @return the password
      */
     public String getPassword() {
-        return password;
+        return connSettings.getPassword();
     }
 
     /**
      * @param password the password to set
      */
     public void setPassword(String password) throws CentralRepoException {
-        if (password == null || password.isEmpty()) {
-            throw new CentralRepoException("Invalid user password. Cannot be empty."); // NON-NLS
-        }
-        this.password = password;
+        connSettings.setPassword(password);
     }
 
-    /**
-     * @return the VALIDATION_QUERY
-     */
-    String getValidationQuery() {
-        return VALIDATION_QUERY;
-    }
-
-    /**
-     * @return the POSTGRES_DRIVER
-     */
-    String getDriver() {
-        return JDBC_DRIVER;
-    }
-
-    /**
-     * @return the JDBC_BASE_URI
-     */
-    String getJDBCBaseURI() {
-        return JDBC_BASE_URI;
+    @Override
+    public DatabaseTestResult testStatus() {
+        if (verifyConnection()) {
+            if (verifyDatabaseExists()) {
+                if (verifyDatabaseSchema()) {
+                    return DatabaseTestResult.TESTEDOK;
```

Comment:
I don't think you named this, but an underscore would really help readability for "TESTEDOK"